packages/react/src/components/SearchField/index.tsx

```diff
@@ -99,6 +101,12 @@ const SearchField = forwardRef<HTMLInputElement, SearchFieldProps>(
             className={classNames(otherProps.className, 'Field__text-input')}
             type="search"
           />
+          {trailingChildren &&
```

Comment:
Nit: We should move this up above to the body:

```jsx
if (trailingChildren === 'string') {
  trailingChildren = <span>{trailingChildren}</span>
}
```

Then we can just render `{trailingChildren}` in the component since `null` or `undefined` values won't render anything.